conda_build/utils.py

```diff
@@ -1514,7 +1514,12 @@ def apply_pin_expressions(version, min_pin='x.x.x.x.x.x.x', max_pin='x'):
             if versions[p_idx][-1] == '.':
                 versions[p_idx] = versions[p_idx][:-1]
     if versions[0]:
-        versions[0] = '>=' + versions[0]
+        if VersionOrder(version) < VersionOrder(versions[0]):
```

Comment:
Maybe
```suggestion
        if (
            VersionOrder(version) < VersionOrder(versions[0])
            and not MatchSpec('pkg ' + version).match(MatchSpec('pkg ' + versions[0]))
        ):
```
could work to handle the `version = '1.2.*'` case?

Reply:
That doesn't handle the case `1.2.3.*`